client/components/search/index.jsx

```diff
@@ -271,15 +272,23 @@ const Search = React.createClass( {
 			spellCheck: 'false'
 		};
 
-		searchClass = classNames( this.props.additionalClasses, {
+		const searchClass = classNames( this.props.additionalClasses, {
 			'is-expanded-to-container': this.props.fitsContainer,
 			'is-open': isOpenUnpinnedOrQueried,
 			'is-searching': this.props.searching,
+			'no-close-button' : this.props.hideClose,
 			rtl: this.props.dir === 'rtl',
 			ltr: this.props.dir === 'ltr',
 			search: true
 		} );
 
+		const inputClass = classNames ( {
+			'search__input': true,
+			'no-close-button': this.props.hideClose,
+			rtl: this.props.dir === 'rtl',
+			ltr: this.props.dir === 'ltr',
```

Comment:
Why do we need to duplicate the `rtl` and `ltr` classes here? (Asking since they're already on L280-281.)